### PR TITLE
improvements: markdown reader

### PR DESCRIPTION
**Pull request by Aminejvm**
Screen Shot 2021-02-18 at 6 38 30 PM

### CODE REVIEW

File: components/system/ThemeProvider.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import { ThemeProvider as EmotionTP } from "@emotion/react";
+
+export default function ThemeProvider({ children }) {
+  const [theme, setTheme] = React.useState({ darkmode: true });
+  const toggleDarkMode = (e) => setTheme((prev) => ({ ...prev, darkmode: e.detail.darkmode }));
+
+  React.useEffect(() => {
+    if (!window) return;
+    window.addEventListener("slate-theme-toggle-darkmode", toggleDarkMode);
```
Comment (jimmylee) [inline components/system/ThemeProvider.js:12]: Can we make this more generic? `set-slate-theme` where it defaults to light.

File: components/core/MarkdownFrame.js [hunk 2]
```diff
@@ -6,15 +6,19 @@ import * as Constants from "~/common/constants";
...
-  padding: 120px calc(32px + 16px + 8px);
-  position: relative;
-  width: 100%;
-  height: 100%;
-  overflow-y: scroll;
-  will-change: transform;
-  color: ${Constants.system.white};
-`;
+const STYLES_ASSET = (theme) => {
+  console.log(theme.darkMode);
```
Comment (jimmylee) [inline components/core/MarkdownFrame.js:10]: Very nice, good job.

File: pages/_app.js [hunk 3]
```diff
@@ -4,16 +4,19 @@ import { Global } from "@emotion/react";
 
 import App from "next/app";
 import { injectGlobalStyles, injectCodeBlockStyles } from "~/common/styles/global";
+import ThemeProvider from "~/components/system/ThemeProvider";
 
 // NOTE(wwwjim):
 // https://nextjs.org/docs/advanced-features/custom-app
 function MyApp({ Component, pageProps }) {
```
Comment (jimmylee) [inline pages/_app.js:11]: Very nice, fix import order please :-)

**Review by jimmylee — APPROVED** [review]
LGTM, if I could get some changes that would be great in a follow up.

File: components/system/ThemeProvider.js (hunk 1 above)
Comment (Aminejvm) [reply components/system/ThemeProvider.js]: You got it, I made it more generic

**Review by jimmylee — APPROVED** [review]
LGTM, this is awesome.

**Pull request merged**